docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,600 @@
+
+Here is one important detail, though: By default, your tags will not be published.
+This requires an additional :command:`git push` with the ``--tags`` option to the
+sibling:
+
+.. code-block:: bash
+
+   $ git push github --tags
+   To https://github.com/adswa/midtermproject.git
+     * [new tag]         ready4analysis -> ready4analysis
+
+
+Yay! Consider your midterm project submitted! Others can now install your
+dataset and check out your data science project -- and even better: they can
+reproduce your data science project easily from scratch!
+
+.. findoutmore:: On the looks and feels of this published dataset
+
+   Now that you have created and published such a YODA-compliant dataset, you
+   are understandably excited how this dataset must look and feel for others.
+   Therefore, you decide to install this dataset in a new location on your
+   computer, just to get a feel for it.
+
+   Replace the ``url`` in the :command:`install` command below with the path
+   to your own ``midtermproject`` GitHub repository:
+
+   .. runrecord:: _examples/DL-101-130-119
+      :language: console
+      :workdir: dl-101/DataLad-101/midterm_project
+
+      $ cd ../../
+      $ datalad install -r --source "https://github.com/adswa/midtermproject.git"
+
+   Note that we performed a *recursive* installation by providing the ``-r``
+   option. Thus, we don't need to install the ``input/`` subdataset anymore.
+   Let's start with the subdataset, and see whether we can retrieve the
+   input ``iris.csv`` file. This should not be a problem, since it's origin
+   is recorded:
+
+
+   .. runrecord:: _examples/DL-101-130-120
+      :language: console
+      :workdir: dl-101
+
+      $ cd midtermproject
+      $ datalad get input/iris.csv
+
+   Nice, this worked well. The output files, however, can not be easily
+   retrieved:
+
+   .. runrecord:: _examples/DL-101-130-121
+      :language: console
+      :workdir: dl-101/midtermproject
+
+      $ datalad get prediction_report.csv pairwise_relationships.png
+
+   Why is that? The file content of these files is managed by Git-annex, and
+   thus only information about the file name and location is known to Git.
+   Because GitHub does not host large data, annexed file content always
+   needs to be deposited somewhere else (e.g., a webserver) to make it
+   accessible via :command:`datalad get`. A later section
+
+   .. todo::
+
+      link 3rd party infra section
+
+   will demonstrate how this can be done. For this dataset, it is not
+   necessary to make the outputs available, though: Because all provenance
+   on their creation was captured, we can simply recompute them with the
+   :command:`datalad rerun` command. Since we tagged the dataset once it was
+   ready for analysis, we can simply rerun any :command:`datalad run` command
+   since this tag:
+
+   .. runrecord:: _examples/DL-101-130-122
+      :language: console
+      :workdir: dl-101/midtermproject
+
+      $ datalad rerun --since ready4analysis
```

Comment:
If tags are not pushed above, this would need to be something else. Ultimately waiting for https://github.com/datalad/datalad/issues/3866

Reply:
For now I am back to rev-parsing the git log...